fixtures/tests/test_fixture.py

```diff
@@ -295,6 +297,29 @@ def _post_process_details(self):
                 {'foo': text_content('foo'),
                  'bar': text_content('oof')}, b.getDetails())
 
+    def test_change_details_in_used_fixture_v2(self):
+        # XXX: This test errors with `'NoneType' object is not iterable`
+        #
+        # I want a way to have a fixture that manipulates the details that it
+        # uses, ideally without subclassing.
+        class A(fixtures.Fixture):
+            def _setUp(self):
+                self.addDetail('foo', text_content('foo'))
+
+        class B(fixtures.Fixture):
+            def _setUp(self):
+                a = self.useFixture(A())
+                self.addDetail('bar', Content(UTF8_TEXT, partial(self._post_process_details, self.getDetails()['foo'])))
+                a.addDetail('foo', text_content('replaced'))
+            def _post_process_details(self, foo):
+                return reversed(foo.as_text())
```

Comment:
FWIW, I take an approach very similar to this in testing-cabal/testtools#171. I think it's much harder to reason about than a cleanup-based approach, because it relies on laziness, and on the byte-provider being called at the exact right time, which is almost impossible to guess without looking at the code.